# Titanic Data Exploration

Pratool Gadtaula

In [13]:
%matplotlib inline

import pandas

df = pandas.read_csv('train.csv')

In [14]:
for el in df.columns:
    print el

PassengerId
Survived
Pclass
Name
Sex
Age
SibSp
Parch
Ticket
Fare
Cabin
Embarked


In [37]:
import thinkplot
import thinkstats2

male_survival = df['Survived'][ df['Sex'] == 'male'].mean()
female_survival = df['Survived'][ df['Sex'] == 'female' ].mean()

print 'Females had a survival rate of %.3f and %.3f of males survived' %(female_survival, male_survival)

Females had a survival rate of 0.742 and 0.189 of males survived


There is a pronounced difference between the male and female survival rate!

In [34]:
print 'Mean age: %.1f years' %df['Age'].mean()
print 'Standard deviation of age: %.2f years' %df['Age'].std()
print 'Young age threshold: %.1f years' %(df['Age'].mean() - df['Age'].std())

Mean age: 29.7 years
Standard deviation of age: 14.53 years
Young age threshold: 15.2 years


I'm going to attempt to figure out the rate of surivival for the young, where a young person is more than 1 standard deviation younger than the mean of the ages. From the Titanic's passengers, that would be an age of 15.2 years.

In [36]:
young = df['Age'].mean() - df['Age'].std()

young_survival = df['Survived'][ df['Age'] < young ].mean()
others_survival = df['Survived'][ df['Age'] >= young].mean()

print 'Youth survival: %.2f' %young_survival
print 'Others survival: %.2f' %others_survival

Youth survival: 0.59
Others survival: 0.38


As I expected, there is a large difference between the youth's survival rate and everyone else's survival rate. Although, this difference is not as large as that between males and females.